### ISSUE TITLE
Revert fancy range behavior.

### ISSUE THREAD

**Issue by andreasplesch** [d3/d3-arrays#5]
While the new range() function is improved, it still produces an extra element for some fractional step sizes.
```
 range(0, 1, 1/49).length;
50
```
Also see https://github.com/mbostock/d3/issues/2524

A fix may have to take into account limited floating point precision, in a similar way to pull request
https://github.com/mbostock/d3/pull/2526

The problem is
```
(1-0)/(1/49)
49.00000000000001
```
which is upped to 50 by Math.ceil().

**Comment by andreasplesch** [comment]
The workaround used in https://github.com/mbostock/d3/issues/2524 still applies:

```
eps=1e-15;
range(0, 64).forEach(function(m){rmax=Math.pow(2,m);
  range(1, 1000).forEach( function(d) {rmax=Math.pow(2,m);
  if (range(0, rmax*(1-eps), rmax/d).length !== d) {console.log(d) ;}})})
```

tested in firefox; still need to test in chrome.

This suggests a similar fix, eg. adjust stop by (1-eps) * stop .

**Comment by andreasplesch** [comment]
Here is a gist which does some checks of the range() function.

http://bl.ocks.org/andreasplesch/ada4b4e89738d4e5d4af

**Comment by andreasplesch** [comment]
Well, I decided to go back to integer ranges.

Perhaps it is worthwhile to add to the documentation that if the number of elements in the generated array is expected to be exactly __n__, then one should use something like
```
// var step = (stop - start)/n;
var step = mystep;
d3.array(0, n).map( function(d) {return start + d * step;};
```
or perhaps provide this as another range function, say d3.nrange(), which takes start, step and number of elements n.

**Comment by mbostock** [comment]
Hmm. I’m tempted to remove the scale hack entirely, since it only works for some cases and makes the behavior harder to understand.

In general, you shouldn’t expect `range(0, 1, 1/n)` to return exactly `n` results because of the limitations of floating point. Using `range(0, n).map(function(i) { return i / n; })` is recommended, and I’m in favor of documenting this recommendation.

**Comment by mbostock** [comment]
Related mbostock/d3#494.

**Comment by andreasplesch** [comment]
Perhaps with the original commit above I now can understand what the conversion to integer is about but it did make the behaviour more opaque. I would say remove it, and perhaps emphasize the limitations a bit more in the documentation.  
It took me a little to figure out that the range() behaviour was the issue when I originally used it for
http://bl.ocks.org/andreasplesch/49b2130b15425c1eebc0/
and got some weird results.
So documentation would be great. I tried to keep the additions in my pull request concise but feel free of course to edit as you feel appropriate. I left the start value in there since for some it may not be too trivial to have seen in an example.

**Comment by mbostock** [comment]
Re-opening this request, because I think you are right that we should remove the fancy behavior.

**Comment by mbostock** [comment]
Fixed in 0.1.2; see d2364d44b6acc977dc6fd0529bfee3097de9488d.
